### PR TITLE
Issue validator warning when beacon node does not use builder

### LINKED ISSUE #11181: Validator registration on non-builder beacon node
Assume the following setups
```
Default: bn1 (w mevboost)     \____ validators w/ builder enabled
Fallback: bn2 (w/o mevboost)  /
```

In the event bn1 goes down, validators switch to bn2. The builder-enabled validators will all default to local EE for block construction.  Validators will see messages such as `could not submit validator registration.` Is that enough? Could we do more on the validator's or the beacon node's end to amplify the intent so validators could notify bn2 to turn on mev-boost asap?

### PR DESCRIPTION
**Pull request by rkapka**
**What type of PR is this?**

UX

**What does this PR do? Why is it needed?**

If the validator uses `--enable-builder`, but the beacon node does not use `--http-mev-relay`, then a warning will be displayed. This also works when failing over to a node without `--http-mev-relay`.

**Which issues(s) does this PR fix?**

Fixes #11181

### CODE REVIEW

File: beacon-chain/builder/service.go [hunk 1]
```diff
@@ -112,7 +115,7 @@ func (s *Service) Status() error {
 		getStatusLatency.Observe(float64(time.Since(start).Milliseconds()))
 	}()
 
-	// Return early if builder isn't initialized in service.
+	// Return early if builder isn't initilized in service.
```
Comment (prestonvanloon) [inline beacon-chain/builder/service.go:118]:
```suggestion
	// Return early if builder isn't initialized in service.
```

File: validator/client/registration.go [hunk 2]
```diff
@@ -30,6 +32,10 @@ func SubmitValidatorRegistrations(
 	if _, err := validatorClient.SubmitValidatorRegistrations(ctx, &ethpb.SignedValidatorRegistrationsV1{
 		Messages: signedRegs,
 	}); err != nil {
+		if strings.Contains(err.Error(), builder.ErrNoBuilder.Error()) {
+			log.Warnln("Beacon node does not utilize a custom builder. Validator registration skipped.")
```
Comment (prestonvanloon) [inline validator/client/registration.go:36]: Is there anything you can add here to give information to the user on how to correct the issue? I.e. "Restart your beacon node with the --mev-relay flag" or something like that

Comment (terencechain) [reply validator/client/registration.go:36]: +1, I think more verbosity will be nice here such as validator keys and such

Comment (rkapka) [reply validator/client/registration.go]: I added the flag name to the log. It impacts all validator keys, so there's no need to add any.

**Review by prestonvanloon — APPROVED** [review]

**Pull request merged**